ext/README.md

```diff
@@ -0,0 +1,49 @@
+# Extensions
+
+The basis of the Extension API is described in a document which should be emulated for all extensions.
+
+## Table of Contents
+
+<!-- toc -->
+* [Table](#table)
+* [Name](#name)
+* [Filename](#filename)
+* [Detail](#detail)
+* [Prior Art](#prior-art)
+<!-- /toc -->
+
+## Table
+
+_notice_: All new `./ext/ext-$name.md` docs MUST be added to this table.
+
+| `$name`  | Definition | Summary |
+|:--:|:--:|:--:|
+| 0 | [ext-0](./ext-0.md) | Base definition of discovering extensions on registry server |
+|  |  |  |
+
+
+## Name
+
+Extension names MUST be unique.
+Names SHOULD include a version.
+
+Each extension's endpoints will be nested below its name.
+
+```HTTP
+    GET /v2/ext/0/...
```

Comment:
the `/v2/ext/` string, in view of the distribution spec, suggests we are reserving `ext` as a logical registry name to be used for registry extension apis. We may need something like `/v3-ext/0' to make that happen?

Reply:
For `registry` level operations under `/v2` in the past we prepended `_`. I think it is best to keep that convention here to not interfere with repository naming.

We could have a wider discussion on the role of `/v2` and the potential to rev that or just make that not required. Personally I think the `/v2` should remain an implementation detail and avoid having the version show up there in the spec. Like other things discussed, I think that belongs as part of the discovery design.

Reply:
note: https://github.com/opencontainers/distribution-spec/blob/master/spec.md#api-version-check

Agree we need long term plan/path for versioning, both for distribution and it's extensions.

Reply:
maybe reserve prepended `_` for registry operations (legacy) and prepend `_ext-[0-9]+` for extensions listed here such as `_ext-0`

Reply:
I think we should try to limit the number of new top-level paths, as they can easily conflict with paths already exposed by registries. I'd prefer if we kept all extensions under the registry root path, whether that be `/v2` or something else moving forward

Reply:
@dmcgowan are you saying you would prefer `/v2/_ext/$name/...`?
@josephschorr is ^ inline with what you're saying?

keep in mind that `$name`-extension would be something like "signatures" for notary-v2, or "events" for the pub-sub proposal.

Reply:
Yeah, `/v2/_ext/...` works. It's not pretty but we can't override the namespace there. I agree with @josephschorr on limiting the number of top level paths. Making them ugly helps I guess ;)

Reply:
1. What would be the guidance be for extension authors to avoid naming collisions/squatting?
Should we consider  `/v2/_ext/oci-artifacts` or `/v2/_ext/signatures` or `/v2/_ext/oci/signatures`.

2. Also would providing guidance around `/v2/_ext` itself. Is there an expectation of returning a list of supported extensions or are the names something that is discovered out of band and we just 404 on `/v2/_ext/oci/signatures`